src/libstd/sync/mpsc/mod.rs

```diff
@@ -1267,6 +1267,24 @@ impl error::Error for TryRecvError {
     }
 }
 
+#[unstable(feature = "mpsc_recv_timeout_error", issue = "0")]
+impl error::Error for RecvTimeoutError {
+    fn description(&self) -> &str {
+        match *self {
+            RecvTimeoutError::Timeout => {
+                "channel is currently empty, but more data may become available later"
```

Comment:
I'd say something like `"timed out waiting on channel"`. The current message doesn't actually say that this is a timeout.